### PR TITLE
Implement basic auth handling

### PR DESCRIPTION
**Pull request by andreaTP**
Depends on:
https://github.com/Apicurio/apicurio-common-app-components/pull/155

### CODE REVIEW

File: lib/auth/ApplicationAuth.tsx [hunk 1]
```diff
@@ -1,61 +1,160 @@
...
+                                id="form-password"
+                                data-testid="basic-auth-login-modal-password"
+                                name="form-password"
+                                value={authState.creds.password}
+                                placeholder="Password"
+                                onChange={onPasswordChange}
+                            />
+                        </FormGroup>
+                    </Form>
+                </Modal>
```
Comment (EricWittmann) [inline lib/auth/ApplicationAuth.tsx:147]: Can you make this a standalone component rather than having it inline?

**Review by EricWittmann — CHANGES_REQUESTED** [review]
Just one request before merging.

File: lib/auth/ApplicationAuth.tsx (hunk 1 above)
Comment (andreaTP) [reply lib/auth/ApplicationAuth.tsx]: sure, make sense, done :slightly_smiling_face:

**Comment by andreaTP** [comment]
@EricWittmann ready for another review round!

File: lib/auth/ApplicationAuth.tsx [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
 import { EmptyState, EmptyStateBody, EmptyStateHeader, EmptyStateIcon, Spinner } from "@patternfly/react-core";
 import { ErrorCircleOIcon } from "@patternfly/react-icons";
-import { AuthService, useAuth } from "./useAuth.ts";
+import { AuthService, UsernameAndPassword, useAuth } from "./useAuth.ts";
 import { If } from "../common";
+import { BasicAuthModal } from "../modals/BasicAuthModal.tsx";
+
+export interface AuthStateData {
+    state: AuthState;
+}
```
Comment (EricWittmann) [inline lib/auth/ApplicationAuth.tsx:10]: Why the wrapper?

**Review by EricWittmann — CHANGES_REQUESTED** [review]
I'm not seeing what these wrapper types are doing for you.

**Review by EricWittmann — APPROVED** [review]
LGTM

**Pull request merged**